framework/harness/mock_endpoints.go

```diff
@@ -154,16 +182,34 @@ func (m *mockEndpointsManager) serveHTTP(w http.ResponseWriter, r *http.Request)
 	e.activeConn = incoming
 	cancellerPtr := &canceller
 	e.cancels = append(e.cancels, cancellerPtr)
+	newConns := e.newConns
 	e.lock.Unlock()
 
+	if newConns == nil {
+		// the endpoint is already closed
+		m.logger.Printf("Received request to already-closed endpoint %s", r.URL)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
 	select { // non-blocking push
-	case e.newConns <- *incoming:
+	case newConns <- *incoming:
 		break
 	default:
 		m.logger.Printf("Incoming connection channel was full for %s", r.URL)
 	}
 
-	e.handler.ServeHTTP(w, transformedReq)
+	wrappedWriter := wrappedResponseWriter{w: w}
+	e.handler.ServeHTTP(&wrappedWriter, transformedReq)
+
+	switch wrappedWriter.status {
+	case http.StatusNotFound:
+		e.logger.Printf("Endpoint %q (%s) received %s request for unrecognized path %s", e.description, e.basePath,
+			r.Method, path)
+	case http.StatusMethodNotAllowed:
+		e.logger.Printf("Endpoint %q (%s) received request with unsupported %s method for path %s", e.description,
+			e.basePath, r.Method, path)
```

Comment:
Just a little extra diagnostic logging for cases where the SDK may have done some kind of incorrect request.